### PR TITLE
Connection tracking split arch changes

### PR DESCRIPTION
**Pull request by hirentha**
Ct Test cases

    Signed-off-by: Hiren Thakkar <hiren.thakkar@intel.com>

### CODE REVIEW

**Comment by hirentha** [comment]
DPDK CT flow dump is not included. once this will merge to main will do that seprately.

File: tests/ct_connection_tap_link_ports_with_scapy_icmp.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+            log.failed(f"FAIL: Verification of icmp packet sent failed with exception {err}")
+
+        log.info("Sending icmp packet: B->A")
+        pkt = simple_icmp_packet(eth_src=self.config_data['traffic']['in_pkt_header']['eth_mac'][1], eth_dst=self.config_data['traffic']['in_pkt_header']['eth_mac'][0], ip_src=self.config_data['traffic']['in_pkt_header']['ip_address'][1] , ip_dst=self.config_data['traffic']['in_pkt_header']['ip_address'][0])
+        
+        send_packet(self, port_ids[self.config_data['traffic']['send_port'][1]], pkt)
+        try:
+            verify_packets(self, pkt, device_number=1, ports=[port_ids[self.config_data['traffic']['receive_port'][1]][1]]) 
+        
+            log.passed(f"PASS: Verification of icmp packets B->A passed, icmp packet received")
```
Comment (sjwahub) [inline tests/ct_connection_tap_link_ports_with_scapy_icmp.py]: In log.passed(), the PASS is not necessary. Please remove it.

File: tests/ct_connection_tap_link_ports_with_scapy_icmp.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        except Exception as err:
+            self.result.addFailure(self, sys.exc_info())
+            log.failed(f"FAIL: Verification of icmp packet B->A sent failed with exception {err}")
```
Comment (sjwahub) [inline tests/ct_connection_tap_link_ports_with_scapy_icmp.py]: The "FAIL" is not necessary. Please remove it.

**Comment by hirentha** [comment]
logs for 11tcs run.
[ct_all_test_logs.txt](https://github.com/ipdk-io/ptf-tests/files/10721948/ct_all_test_logs.txt)

File: tests/ct_connection_tap_link_ports_with_scapy_udp.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            log.failed(f"FAIL: Verification of udp packet sent failed with exception {err}")
+
+        log.info("Sending udp packet: B->A")
+        pkt = simple_udp_packet(eth_src=self.config_data['traffic']['in_pkt_header']['eth_mac'][1], eth_dst=self.config_data['traffic']['in_pkt_header']['eth_mac'][0], ip_src=self.config_data['traffic']['in_pkt_header']['ip_address'][1] , ip_dst=self.config_data['traffic']['in_pkt_header']['ip_address'][0], udp_sport=self.config_data['traffic']['in_pkt_header']['udp_port'][1], udp_dport=self.config_data['traffic']['in_pkt_header']['udp_port'][0])
+        
+        send_packet(self, port_ids[self.config_data['traffic']['send_port'][1]], pkt)
+        try:
+            verify_packets(self, pkt, device_number=0, ports=[port_ids[self.config_data['traffic']['receive_port'][1]][1]]) 
+        
+            log.passed(f"PASS: Verification of udp packets B->A passed, udp packet received")
```
Comment (sjwahub) [inline tests/ct_connection_tap_link_ports_with_scapy_udp.py]: remove the word PASS in log.passed()

File: tests/ct_connection_tap_link_ports_with_scapy_udp.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        except Exception as err:
+            self.result.addFailure(self, sys.exc_info())
+            log.failed(f"FAIL: Verification of udp packet B->A sent failed with exception {err}")
```
Comment (sjwahub) [inline tests/ct_connection_tap_link_ports_with_scapy_udp.py]: please remove FAIL in log.failed()

File: tests/ct_connection_vhost_ports_with_netperf_udp.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+import ptf.dataplane as dataplane
+from ptf.base_tests import BaseTest
+from ptf.testutils import *
+from ptf import config
+
+# scapy related imports
+from scapy.packet import *
+from scapy.fields import *
+from scapy.all import *
+
```
Comment (sjwahub) [inline tests/ct_connection_vhost_ports_with_netperf_udp.py]: The line 34-37 scapy are never used. Only netperf traffic. Suggest to remove them for clean code.

File: tests/ct_connection_vhost_ports_with_ping_icmp.py [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+import ptf
+import ptf.dataplane as dataplane
+from ptf.base_tests import BaseTest
+from ptf.testutils import *
+from ptf import config
+
+# scapy related imports
+from scapy.packet import *
+from scapy.fields import *
+from scapy.all import *
```
Comment (sjwahub) [inline tests/ct_connection_vhost_ports_with_ping_icmp.py]: The line 34 -37 are never used. Suggest to remove them for clean code.

File: tests/ct_connection_vhost_ports_with_netperf_udp.py [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+    def setUp(self):
+        BaseTest.setUp(self)
+        self.result = unittest.TestResult()
+        config["relax"] = True # for verify_packets to ignore other packets received at the interface
+        
+        test_params = test_params_get()
+        config_json = test_params['config_json']
+        self.dataplane = ptf.dataplane_instance
+        ptf.dataplane_instance = ptf.dataplane.DataPlane(config)
```
Comment (sjwahub) [inline tests/ct_connection_vhost_ports_with_netperf_udp.py]: As no scapy involved, suggest to remove line 57-58

File: tests/ct_connection_vhost_ports_with_netperf_udp.py [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+            self.fail("Failed to configure gnmi ctl ports")
+
+
+        # get port list and add to dataplane
+        port_list = self.config_data['port_list']
+        port_ids = test_utils.add_port_to_dataplane(port_list)
+        
+        for port_id, ifname in config["port_map"].items():
+            device, port = port_id
+            self.dataplane.port_add(ifname, device, port)
```
Comment (sjwahub) [inline tests/ct_connection_vhost_ports_with_netperf_udp.py]: As no scapy involved, it's redundant to add port into dataplane. Suggest to remove line 77-79

File: tests/ct_connection_vhost_ports_with_netperf_udp.py [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+        
+        #send netperf from local VM
+        i=1
+        log.info(f"Initially execute netperf on VM{i}")
+        if not test_utils.vm_netperf_client(self.conn_obj_list[i], self.config_data['vm'][i]['remote_ip'],
+                self.config_data['netperf']['testlen'], self.config_data['netperf']['testname'], option = self.config_data['netperf']['cmd_option']):
+            self.result.addFailure(self, sys.exc_info())
+            self.fail(f"FAIL: failed to start netperf")
+
+        self.dataplane.kill()
```
Comment (sjwahub) [inline tests/ct_connection_vhost_ports_with_netperf_udp.py]: With previous comment, as no scapy involved, dataplane related code is removed. This line is needed to remove too.

File: tests/ct_connection_vhost_ports_with_ping_icmp.py [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+
+    def setUp(self):
+        BaseTest.setUp(self)
+        self.result = unittest.TestResult()
+        config["relax"] = True # for verify_packets to ignore other packets received at the interface
+        
+        test_params = test_params_get()
+        config_json = test_params['config_json']
+        self.dataplane = ptf.dataplane_instance
+        ptf.dataplane_instance = ptf.dataplane.DataPlane(config)
```
Comment (sjwahub) [inline tests/ct_connection_vhost_ports_with_ping_icmp.py]: In this case, only ping is used, no dataplane traffic involved. Suggest to remove line 57-58.

File: tests/ct_connection_vhost_ports_with_ping_icmp.py [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+            self.fail("Failed to configure gnmi ctl ports")
+
+
+        # get port list and add to dataplane
+        port_list = self.config_data['port_list']
+        port_ids = test_utils.add_port_to_dataplane(port_list)
+        
+        for port_id, ifname in config["port_map"].items():
+            device, port = port_id
+            self.dataplane.port_add(ifname, device, port)
```
Comment (sjwahub) [inline tests/ct_connection_vhost_ports_with_ping_icmp.py]: In this case, only ping is used, no dataplane traffic involved. Suggest to remove line 77-79.

File: tests/ct_connection_vhost_ports_with_ping_icmp.py [hunk 12]
```diff
@@ -0,0 +1,165 @@
...
+        result = test_utils.vm_to_vm_ping_test(
+            conn2, self.config_data["vm"][1]["remote_ip"]
+        )
+        if not result:
+            self.result.addFailure(self, sys.exc_info())
+            log.failed("Ping test failed for VM1")
+
+
+
+        self.dataplane.kill()
```
Comment (sjwahub) [inline tests/ct_connection_vhost_ports_with_ping_icmp.py]: Same reason. Suggest to remove this line

File: tests/ct_multiple_connection_tap_link_ports.py [hunk 13]
```diff
@@ -80,318 +81,342 @@ def runTest(self):
...
+        pkt = simple_tcp_packet(eth_src=self.config_data['traffic']['in_pkt_header']['eth_mac'][2], eth_dst=self.config_data['traffic']['in_pkt_header']['eth_mac'][3], ip_src=self.config_data['traffic']['in_pkt_header']['ip_address'][2] , ip_dst=self.config_data['traffic']['in_pkt_header']['ip_address'][3], tcp_sport=self.config_data['traffic']['in_pkt_header']['tcp_port'][2], tcp_dport=self.config_data['traffic']['in_pkt_header']['tcp_port'][3], tcp_flags="A")
         send_packet(self, port_ids[self.config_data['traffic']['send_port'][2]], pkt)
         try:
             verify_no_packet(self, pkt, port_ids[self.config_data['traffic']['receive_port'][2]][1])
 
-            print(f"PASS: Verification of data packet check passed : No traffic between A->B")
+            log.passed(f"PASS: Verification of data packet check passed : No traffic between A->B")
```
Comment (sjwahub) [inline tests/ct_multiple_connection_tap_link_ports.py]: The PASS need to remove from log.passed()

File: tests/ct_multiple_connection_tap_link_ports.py [hunk 14]
```diff
@@ -80,318 +81,342 @@ def runTest(self):
...
         except Exception as err:
             self.result.addFailure(self, sys.exc_info())
-            print(f"FAIL: Verification of data packet sent failed with exception {err}: A->B")
+            log.failed(f"FAIL: Verification of data packet sent failed with exception {err}: A->B")
```
Comment (sjwahub) [inline tests/ct_multiple_connection_tap_link_ports.py]: The "FAIL" is needed to remove from log.failed()

File: tests/ct_multiple_connection_tap_link_ports.py [hunk 15]
```diff
@@ -80,318 +81,342 @@ def runTest(self):
...
+        log.info("Sending data packet: B->A")
+        pkt = simple_tcp_packet(eth_src=self.config_data['traffic']['in_pkt_header']['eth_mac'][3], eth_dst=self.config_data['traffic']['in_pkt_header']['eth_mac'][2], ip_src=self.config_data['traffic']['in_pkt_header']['ip_address'][3] , ip_dst=self.config_data['traffic']['in_pkt_header']['ip_address'][2], tcp_sport=self.config_data['traffic']['in_pkt_header']['tcp_port'][3], tcp_dport=self.config_data['traffic']['in_pkt_header']['tcp_port'][2], tcp_flags="A")
         send_packet(self, port_ids[self.config_data['traffic']['send_port'][3]], pkt)
         try:
             verify_no_packet(self, pkt, port_ids[self.config_data['traffic']['receive_port'][3]][1])
 
-            print(f"PASS: Verification of date packet check passed : No traffic between B->A")
+            log.passed(f"PASS: Verification of date packet check passed : No traffic between B->A")
```
Comment (sjwahub) [inline tests/ct_multiple_connection_tap_link_ports.py]: The PASS is needed removed from log.passed()

File: tests/ct_multiple_connection_tap_link_ports.py [hunk 16]
```diff
@@ -80,318 +81,342 @@ def runTest(self):
...
         except Exception as err:
             self.result.addFailure(self, sys.exc_info())
-            print(f"FAIL: Verification of data packet sent failed with exception {err}: B->A")
+            log.failed(f"FAIL: Verification of data packet sent failed with exception {err}: B->A")
```
Comment (sjwahub) [inline tests/ct_multiple_connection_tap_link_ports.py]: The "FAIL" needed to remove from log.failed()

File: tests/ct_multiple_connection_tap_link_ports.py [hunk 17]
```diff
@@ -80,318 +81,342 @@ def runTest(self):
...
-        print("sending SYN packet: A->B")
+        log.info("Scenario : Connection Establishment for SYN TAP0 and Link Port")
+        log.info("--------------------------------------------------------------")
+        log.info("sending SYN packet: A->B")
         pkt = simple_tcp_packet(eth_src=self.config_data['traffic']['in_pkt_header']['eth_mac'][0], eth_dst=self.config_data['traffic']['in_pkt_header']['eth_mac'][1], ip_src=self.config_data['traffic']['in_pkt_header']['ip_address'][0] , ip_dst=self.config_data['traffic']['in_pkt_header']['ip_address'][1], tcp_sport=self.config_data['traffic']['in_pkt_header']['tcp_port'][0], tcp_dport=self.config_data['traffic']['in_pkt_header']['tcp_port'][1])
         send_packet(self, port_ids[self.config_data['traffic']['send_port'][0]], pkt)
         try:
             verify_packets(self, pkt, device_number=0, ports=[port_ids[self.config_data['traffic']['receive_port'][0]][1]])              
-            print(f"PASS: Verification of packets passed, Syn packet received")
+            log.passed(f"PASS: Verification of packets passed, Syn packet received")
```
Comment (sjwahub) [inline tests/ct_multiple_connection_tap_link_ports.py]: The PASS needed to remove from log.passed()

File: tests/ct_multiple_connection_tap_link_ports.py [hunk 18]
```diff
@@ -80,318 +81,342 @@ def runTest(self):
...
         except Exception as err:
             self.result.addFailure(self, sys.exc_info())
-            print(f"FAIL: Verification of Syn packet sent failed with exception {err}")
+            log.failed(f"FAIL: Verification of Syn packet sent failed with exception {err}")
```
Comment (sjwahub) [inline tests/ct_multiple_connection_tap_link_ports.py]: The FAIL needed to remove from log.failed()

File: tests/ct_multiple_connection_tap_link_ports.py [hunk 19]
```diff
@@ -80,318 +81,342 @@ def runTest(self):
...
-        print("sending SYN packet: A->B")
+        log.info("Scenario : Connection Establishment for SYN TAP1 and TAP2 Port")
+        log.info("--------------------------------------------------------------")
+        log.info("sending SYN packet: A->B")
         pkt = simple_tcp_packet(eth_src=self.config_data['traffic']['in_pkt_header']['eth_mac'][2], eth_dst=self.config_data['traffic']['in_pkt_header']['eth_mac'][3], ip_src=self.config_data['traffic']['in_pkt_header']['ip_address'][2] , ip_dst=self.config_data['traffic']['in_pkt_header']['ip_address'][3], tcp_sport=self.config_data['traffic']['in_pkt_header']['tcp_port'][2], tcp_dport=self.config_data['traffic']['in_pkt_header']['tcp_port'][3])
         send_packet(self, port_ids[self.config_data['traffic']['send_port'][2]], pkt)
         try:
             verify_packets(self, pkt, device_number=0, ports=[port_ids[self.config_data['traffic']['receive_port'][2]][1]])
-            print(f"PASS: Verification of packets passed, Syn packet received")
+            log.passed(f"PASS: Verification of packets passed, Syn packet received")
```
Comment (sjwahub) [inline tests/ct_multiple_connection_tap_link_ports.py]: The PASS needed to remove from log.passed()

File: tests/ct_multiple_connection_tap_link_ports.py [hunk 20]
```diff
@@ -80,318 +81,342 @@ def runTest(self):
...
         except Exception as err:
             self.result.addFailure(self, sys.exc_info())
-            print(f"FAIL: Verification of Syn packet sent failed with exception {err}")
+            log.failed(f"FAIL: Verification of Syn packet sent failed with exception {err}")
```
Comment (sjwahub) [inline tests/ct_multiple_connection_tap_link_ports.py]: The FAIL needed to remove from log.failed()

File: tests/ct_multiple_connection_tap_link_ports.py [hunk 21]
```diff
@@ -80,318 +81,342 @@ def runTest(self):
...
+        log.info("--------------------------------------------------------------")
+        log.info("Sending SYN+ACK packet: B->A")
         pkt = simple_tcp_packet(eth_src=self.config_data['traffic']['in_pkt_header']['eth_mac'][1], eth_dst=self.config_data['traffic']['in_pkt_header']['eth_mac'][0], ip_src=self.config_data['traffic']['in_pkt_header']['ip_address'][1] , ip_dst=self.config_data['traffic']['in_pkt_header']['ip_address'][0], tcp_sport=self.config_data['traffic']['in_pkt_header']['tcp_port'][1], tcp_dport=self.config_data['traffic']['in_pkt_header']['tcp_port'][0], tcp_flags="SA")
         
         send_packet(self, port_ids[self.config_data['traffic']['send_port'][1]], pkt)
         try:
             verify_packets(self, pkt, device_number=0, ports=[port_ids[self.config_data['traffic']['receive_port'][1]][1]]) 
         
-            print(f"PASS: Verification of packets passed, SynAck packet received")
+            log.passed(f"PASS: Verification of packets passed, SynAck packet received")
```
Comment (sjwahub) [inline tests/ct_multiple_connection_tap_link_ports.py]: The PASS needed to remove from log.passed()

File: tests/ct_multiple_connection_tap_link_ports.py [hunk 22]
```diff
@@ -80,318 +81,342 @@ def runTest(self):
...
         except Exception as err:
             self.result.addFailure(self, sys.exc_info())
-            print(f"FAIL: Verification of SynAck packet sent failed with exception {err}")
+            log.failed(f"FAIL: Verification of SynAck packet sent failed with exception {err}")
```
Comment (sjwahub) [inline tests/ct_multiple_connection_tap_link_ports.py]: The FAIL needed to remove from log.failed()

File: tests/ct_multiple_connection_tap_link_ports.py [hunk 23]
```diff
@@ -80,318 +81,342 @@ def runTest(self):
...
 
-        print("Sending ACK packet: A->B")
+        log.info("Sending ACK packet: A->B")
         pkt = simple_tcp_packet(eth_src=self.config_data['traffic']['in_pkt_header']['eth_mac'][0], eth_dst=self.config_data['traffic']['in_pkt_header']['eth_mac'][1], ip_src=self.config_data['traffic']['in_pkt_header']['ip_address'][0] , ip_dst=self.config_data['traffic']['in_pkt_header']['ip_address'][1], tcp_sport=self.config_data['traffic']['in_pkt_header']['tcp_port'][0], tcp_dport=self.config_data['traffic']['in_pkt_header']['tcp_port'][1], tcp_flags="A")
         send_packet(self, port_ids[self.config_data['traffic']['send_port'][0]], pkt)
         try:
 
             verify_packets(self, pkt, device_number=0, ports=[port_ids[self.config_data['traffic']['receive_port'][0]][1]])  
-            print(f"PASS: Verification of packets passed, ACK packet received")
+            log.passed(f"PASS: Verification of packets passed, ACK packet received")
```
Comment (sjwahub) [inline tests/ct_multiple_connection_tap_link_ports.py]: The PASS needed to remove from log.passed()

File: tests/ct_multiple_connection_tap_link_ports.py [hunk 24]
```diff
@@ -80,318 +81,342 @@ def runTest(self):
...
         except Exception as err:
             self.result.addFailure(self, sys.exc_info())
-            print(f"FAIL: Verification of ACK packets sent failed with exception {err}")
+            log.failed(f"FAIL: Verification of ACK packets sent failed with exception {err}")
```
Comment (sjwahub) [inline tests/ct_multiple_connection_tap_link_ports.py]: The FAIL need to remove from log.failed()

File: tests/ct_multiple_connection_tap_link_ports.py [hunk 25]
```diff
@@ -80,318 +81,342 @@ def runTest(self):
...
+        log.info("Scenario : Connection Establishment for SYN+ACK TAP1 and TAP2 Port")
+        log.info("------------------------------------------------------------------")
+        log.info("Sending SYN+ACK packet: B->A")
         pkt = simple_tcp_packet(eth_src=self.config_data['traffic']['in_pkt_header']['eth_mac'][3], eth_dst=self.config_data['traffic']['in_pkt_header']['eth_mac'][2], ip_src=self.config_data['traffic']['in_pkt_header']['ip_address'][3] , ip_dst=self.config_data['traffic']['in_pkt_header']['ip_address'][2], tcp_sport=self.config_data['traffic']['in_pkt_header']['tcp_port'][3], tcp_dport=self.config_data['traffic']['in_pkt_header']['tcp_port'][2], tcp_flags="SA")
         send_packet(self, port_ids[self.config_data['traffic']['send_port'][3]], pkt)
         try:
             verify_packets(self, pkt, device_number=0, ports=[port_ids[self.config_data['traffic']['receive_port'][3]][1]])
 
-            print(f"PASS: Verification of packets passed, SynAck packet received")
+            log.passed(f"PASS: Verification of packets passed, SynAck packet received")
```
Comment (sjwahub) [inline tests/ct_multiple_connection_tap_link_ports.py]: The PASS need to remove from log.passed()

File: tests/ct_multiple_connection_tap_link_ports.py [hunk 26]
```diff
@@ -80,318 +81,342 @@ def runTest(self):
...
         except Exception as err:
             self.result.addFailure(self, sys.exc_info())
-            print(f"FAIL: Verification of SynAck packet sent failed with exception {err}")
+            log.failed(f"FAIL: Verification of SynAck packet sent failed with exception {err}")
```
Comment (sjwahub) [inline tests/ct_multiple_connection_tap_link_ports.py]: The FAIL need to remove from log.failed()

File: tests/ct_multiple_connection_tap_link_ports.py [hunk 27]
```diff
@@ -80,318 +81,342 @@ def runTest(self):
...
-        print("Sending ACK packet: A->B")
+        log.info("Sending ACK packet: A->B")
         pkt = simple_tcp_packet(eth_src=self.config_data['traffic']['in_pkt_header']['eth_mac'][2], eth_dst=self.config_data['traffic']['in_pkt_header']['eth_mac'][3], ip_src=self.config_data['traffic']['in_pkt_header']['ip_address'][2] , ip_dst=self.config_data['traffic']['in_pkt_header']['ip_address'][3], tcp_sport=self.config_data['traffic']['in_pkt_header']['tcp_port'][2], tcp_dport=self.config_data['traffic']['in_pkt_header']['tcp_port'][3], tcp_flags="A")
         send_packet(self, port_ids[self.config_data['traffic']['send_port'][2]], pkt)
         try:
 
             verify_packets(self, pkt, device_number=0, ports=[port_ids[self.config_data['traffic']['receive_port'][2]][1]])
 
-            print(f"PASS: Verification of packets passed, ACK packet received")
+            log.passed(f"PASS: Verification of packets passed, ACK packet received")
```
Comment (sjwahub) [inline tests/ct_multiple_connection_tap_link_ports.py]: The PASS need to remove from log.passed()

File: tests/ct_multiple_connection_tap_link_ports.py [hunk 28]
```diff
@@ -80,318 +81,342 @@ def runTest(self):
...
         except Exception as err:
             self.result.addFailure(self, sys.exc_info())
-            print(f"FAIL: Verification of ACK packets sent failed with exception {err}") 
+            log.failed(f"FAIL: Verification of ACK packets sent failed with exception {err}")
```
Comment (sjwahub) [inline tests/ct_multiple_connection_tap_link_ports.py:401]: The FAIL need to remove from log.failed()

File: tests/ct_multiple_connection_tap_link_ports.py [hunk 29]
```diff
@@ -80,318 +81,342 @@ def runTest(self):
...
+        pkt = simple_tcp_packet(eth_src=self.config_data['traffic']['in_pkt_header']['eth_mac'][0], eth_dst=self.config_data['traffic']['in_pkt_header']['eth_mac'][1], ip_src=self.config_data['traffic']['in_pkt_header']['ip_address'][0] , ip_dst=self.config_data['traffic']['in_pkt_header']['ip_address'][1], tcp_sport=self.config_data['traffic']['in_pkt_header']['tcp_port'][0], tcp_dport=self.config_data['traffic']['in_pkt_header']['tcp_port'][1], tcp_flags="A")
         send_packet(self, port_ids[self.config_data['traffic']['send_port'][0]], pkt)
         try:
             verify_packets(self, pkt, device_number=0, ports=[port_ids[self.config_data['traffic']['receive_port'][0]][1]])  
-            print(f"PASS: Verification of data packet passed : A->B")
+            log.passed(f"PASS: Verification of data packet passed : A->B")
```
Comment (sjwahub) [inline tests/ct_multiple_connection_tap_link_ports.py]: The PASS need to remove from log.passed()

File: tests/ct_multiple_connection_tap_link_ports.py [hunk 30]
```diff
@@ -80,318 +81,342 @@ def runTest(self):
...
         except Exception as err:
             self.result.addFailure(self, sys.exc_info())
-            print(f"FAIL: Verification of data packet sent failed with exception {err}: A->B")
+            log.failed(f"FAIL: Verification of data packet sent failed with exception {err}: A->B")
```
Comment (sjwahub) [inline tests/ct_multiple_connection_tap_link_ports.py]: The FAIL need to remove from log.failed()

**Comment by sjwahub** [comment]
1. The word PASS and FAIL need to remove from log.passed() and log.failed() respectfully. 
2. New test cases needed to add into test_to_run.txt

File: pre_test.sh [hunk 31]
```diff
@@ -53,6 +53,10 @@ $IPDK_RECIPE/install/sbin/ovs-vswitchd --detach --no-chdir unix:$RUN_OVS/var/run
 echo "set hugepages"
 $IPDK_RECIPE/scripts/dpdk/set_hugepages.sh
 
+rm -rf /usr/share/stratum/dpdk/dpdk_skip_p4.conf
+cp common/p4c_artifacts/pna_tcp_connection_tracking/dpdk_skip_p4.conf /usr/share/stratum/dpdk/
```
Comment (dkaranjai) [inline pre_test.sh:57]: /usr/share/stratum/dpdk - will this path always exist? or do you want to check if it exists, if not then create it?

File: tests/ct_connection_tap_link_ports_with_scapy_icmp.py [hunk 32]
```diff
@@ -0,0 +1,143 @@
...
+        self.dataplane = ptf.dataplane_instance
+        ptf.dataplane_instance = ptf.dataplane.DataPlane(config)
+        self.capture_port = test_params['pci_bdf'][:-1] + "1"
+        self.config_data = get_config_dict(config_json, test_params['pci_bdf'])
+        self.gnmictl_params = get_gnmi_params_simple(self.config_data)
+        self.interface_ip_list = get_interface_ipv4_dict(self.config_data)
+
+
+    def runTest(self):
+        if not test_utils.gen_dep_files_p4c_dpdk_pna_tdi_pipeline_builder(self.config_data):
```
Comment (dkaranjai) [inline tests/ct_connection_tap_link_ports_with_scapy_icmp.py]: step comment missing

File: tests/ct_connection_tap_link_ports_with_scapy_icmp.py [hunk 33]
```diff
@@ -0,0 +1,143 @@
...
+            self.result.addFailure(self, sys.exc_info())
+            self.fail("Failed to generate P4C artifacts or pb.bin")
+        
+        if not gnmi_ctl_set_and_verify(self.gnmictl_params):
```
Comment (dkaranjai) [inline tests/ct_connection_tap_link_ports_with_scapy_icmp.py:81]: step comment missing

File: tests/ct_connection_tap_link_ports_with_scapy_icmp.py [hunk 34]
```diff
@@ -0,0 +1,143 @@
...
+        # get port list and add to dataplane
+        port_list = self.config_data['port_list']
+        port_list[0] = test_utils.get_port_name_from_pci_bdf(self.capture_port)
+        port_ids = test_utils.add_port_to_dataplane(port_list)
+        
+        for port_id, ifname in config["port_map"].items():
+            device, port = port_id
+            self.dataplane.port_add(ifname, device, port)
+
+        if not p4rt_ctl.p4rt_ctl_set_pipe(self.config_data['switch'], self.config_data['pb_bin'], self.config_data['p4_info']):
```
Comment (dkaranjai) [inline tests/ct_connection_tap_link_ports_with_scapy_icmp.py]: step comment missing

File: tests/ct_connection_tap_link_ports_with_scapy_icmp.py [hunk 35]
```diff
@@ -0,0 +1,143 @@
...
+            self.result.addFailure(self, sys.exc_info())
+            self.fail("Failed to set pipe")
+
+        table = self.config_data['table'][0]
```
Comment (dkaranjai) [inline tests/ct_connection_tap_link_ports_with_scapy_icmp.py]: step comment missing

File: tests/ct_connection_tap_link_ports_with_scapy_icmp.py [hunk 36]
```diff
@@ -0,0 +1,143 @@
...
+        # Sleep 5 Sec before udp connection establishment
+        time.sleep(5)
+            
+        log.info("Scenario : Connection Establishment")
+        log.info("sending icmp packet: A->B")
+        pkt = simple_icmp_packet(eth_src=self.config_data['traffic']['in_pkt_header']['eth_mac'][0], eth_dst=self.config_data['traffic']['in_pkt_header']['eth_mac'][1], ip_src=self.config_data['traffic']['in_pkt_header']['ip_address'][0] , ip_dst=self.config_data['traffic']['in_pkt_header']['ip_address'][1])
+        send_packet(self, port_ids[self.config_data['traffic']['send_port'][0]], pkt)
+        try:
+            verify_packets(self, pkt, device_number=1, ports=[port_ids[self.config_data['traffic']['receive_port'][0]][1]])              
+            log.passed(f"PASS: Verification of icmp packets A->B passed, icmp packet received")
```
Comment (dkaranjai) [inline tests/ct_connection_tap_link_ports_with_scapy_icmp.py]: remove PASS:

**Review by dkaranjai — CHANGES_REQUESTED** [review]
General comments:
1. Step comment missing in multiple cases
2. log.passed and log.failed dont need "PASS" and "FAIL" in the message
3. use "black" tool once to reformat the code

**Review by stadju — COMMENTED** [review]
In connection test, after FIN and RST, maybe you can send data packet to make sure the connection is closed.
2/13/2023,23:18:59  INFO: Sending FIN packet: A->B
2/13/2023,23:18:59  PASSED: PASS: Verification of packets passed, FIN packet received
2/13/2023,23:18:59  INFO: Sending FIN+ACK packet: B->A
2/13/2023,23:18:59  PASSED: PASS: Verification of packets passed, FIN+ACK packet received
2/13/2023,23:18:59  INFO: Sending FIN packet: B->A
2/13/2023,23:18:59  PASSED: PASS: Verification of packets passed, FIN packet received B->A
2/13/2023,23:18:59  INFO: Sending ACK packet: A->B
2/13/2023,23:18:59  PASSED: PASS: Verification of packets passed, ACK packet received
===> send data packet
2/13/2023,23:18:59  INFO: Connection Reset
2/13/2023,23:18:59  INFO: Sending RST packet: A->B
2/13/2023,23:18:59  PASSED: PASS: Verification of packets passed, RST packet received
===> send data packet

**Comment by hirentha** [comment]
Addressed all comments and amended all changes. Please review it. 

====================
Summary
====================
ct_single_connection_tap_link_ports : PASSED
ct_multiple_connection_tap_link_ports : PASSED
ct_rule_del_add_test : PASSED
nt_ct_unsupported_arguments_values : PASSED
ct_single_connection_vhost_ports_with_netperf : PASSED
ct_multiple_connection_vhost_ports_with_netperf : PASSED
ct_timer_test : PASSED
ct_connection_tap_link_ports_with_scapy_icmp : PASSED
ct_connection_tap_link_ports_with_scapy_udp : PASSED
ct_connection_vhost_ports_with_ping_icmp : PASSED
ct_connection_vhost_ports_with_netperf_udp : PASSED

[ct_all_test_logs.txt](https://github.com/ipdk-io/ptf-tests/files/10730676/ct_all_test_logs.txt)

**Comment by dkaranjai** [comment]
Applied formatting of code using black

**Review by dkaranjai — APPROVED** [review]
LGTM

**Review by sjwahub — APPROVED** [review]

**Pull request merged**